crates/kitsune_p2p/kitsune_p2p/tests/sharding.rs

```diff
@@ -42,120 +45,134 @@ async fn publish_to_basis_from_inside() {
         params
     };
 
-    // Arcs are this long by default, with an adjustment to ensure overlap.
-    let base_len = u32::MAX / 5;
-
-    let dim = SpaceDimension::standard();
-
     let sender_idx = 3;
     let should_recv_idx = 2;
 
-    let mut agents = Vec::new();
-    let mut accepted_agent_setup = false;
-
-    let mut chosen_basis = None;
-
-    'agent_setup: for _ in 0..10 {
-        agents.clear();
-        for i in 0..5 {
-            let mut harness = KitsuneTestHarness::try_new("")
-                .await
-                .expect("Failed to setup test harness")
-                .configure_tx5_network(signal_url)
-                .use_bootstrap_server(bootstrap_addr)
-                .update_tuning_params(tuner);
-
-            let sender = harness.spawn().await.expect("should be able to spawn node");
-
-            let mut agent = harness.create_agent().await;
-            let mut found_loc = false;
-            for _ in 0..1000 {
-                let loc = agent.get_loc().as_();
-                // Search from the start of our agent, up to halfway through it. Agents that are in the
-                // upper part of their range are less likely to overlap with the previous agent.
-                if (loc) > base_len * i && (loc as f64) < base_len as f64 * (i as f64 + 0.5) {
-                    found_loc = true;
-                    break;
+    let agents = setup_overlapping_agents(
```

Comment:
Existing tests haven't changed, I've just extracted some code to reduce duplication

Reply:
Are you saying that we don't really need to review what follows in this file?

Reply:
The existing tests haven't changed, but the PR is adding new tests below. Those are important and I do need a review for them please. Was just trying to help out with this diff at the top of the file - wasn't clear enough!

Reply:
@ThetaSinner it's hard to tell what's new and what's changed from the diff, could you point out which tests are new?

Reply:
Oh right, yeah that diff is really unclear because the common setup I extracted has got mixed in with the last test.

The new tests are `gossip_to_basis_from_inside` and `gossip_to_basis_from_outside`